File: extensions/live-activity/native/LiveActivityModule/LiveActivityModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LiveActivityModule, NSObject)

RCT_EXTERN_METHOD(start:(NSDictionary *)params
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(update:(NSString *)activityId
                  heading:(NSString *)heading)

RCT_EXTERN_METHOD(end:(NSString *)activityId)

@end

File: extensions/live-activity/native/LiveActivityModule/LiveActivityModule.swift
// LiveActivityModule.swift
import Foundation
import React

@objc(LiveActivityModule)
class LiveActivityModule: NSObject {

  @objc
  func start(_ params: NSDictionary,
             resolver resolve: @escaping RCTPromiseResolveBlock,
             rejecter reject: @escaping RCTPromiseRejectBlock) {

    guard #available(iOS 16.1, *) else {
      reject("UNSUPPORTED", "Live Activities are only available on iOS 16.1 and above", nil)
      return
    }

    ActivityKitWrapper.shared.start(params: params, resolve: resolve, reject: reject)
  }

  @objc
  func update(_ activityId: String, heading: String) {
    guard #available(iOS 16.1, *) else {
      return
    }

    ActivityKitWrapper.shared.update(activityId: activityId, heading: heading)
  }

  @objc
  func end(_ activityId: String) {
    guard #available(iOS 16.1, *) else {
      return
    }

    ActivityKitWrapper.shared.end(activityId: activityId)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }
}
